nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/memory/MemoryDirectorySession.java

```diff
@@ -124,7 +121,8 @@ public DocumentModel createEntryWithoutReferences(Map<String, Object> fieldMap)
         String id = String.valueOf(rawId);
         Map<String, Object> map = data.get(id);
         if (map != null) {
-            throw new DirectoryException(String.format("Entry with id %s already exists", id));
+            throw new DirectoryException(
+                    String.format("Entry with id %s already exists in directory '%s'", id, directory.getName()));
```

Comment:
Could you make consistent the parameters printing? (ie: surround each parameters by `'` or don't).

Reply:
i like to add ' to let the message more clear. But in this case changing the first one may be broken some client/ps code in the case where they check the exception message like in our tests. at the end i remove the caractere ' on directory name.